### PR TITLE
feat(operation): adapt AddData & UpdateData operation with upload_files

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-29xepr6 对接上传二进制文件与其他表格数据](https://app.clickup.com/t/29xepr6)

File: graviti/operation/frame.py [hunk 1]
```diff
@@ -123,8 +138,22 @@ def do(  # pylint: disable=invalid-name
             sheet: The sheet name.
 
         """
-        data = self.data.to_pylist()
-        for batch in chunked(data, _MAX_BATCH_SIZE):
+        dataframe = self.data
+        arrays = _get_file_arrays(dataframe.schema, dataframe)  # type: ignore[arg-type]
+        data = dataframe.to_pylist()
+
+        for batch, *file_arrays in chunked(zip(data, *arrays), _MAX_BATCH_SIZE):
```
Comment (linjiX) [inline graviti/operation/frame.py:145]: will it be clear to separate binary file upload and add data into two for loop？

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/operation/frame.py [hunk 2]
```diff
@@ -123,8 +138,22 @@ def do(  # pylint: disable=invalid-name
             sheet: The sheet name.
 
         """
-        data = self.data.to_pylist()
-        for batch in chunked(data, _MAX_BATCH_SIZE):
+        dataframe = self.data
+        arrays = _get_file_arrays(dataframe.schema, dataframe)  # type: ignore[arg-type]
```
Comment (linjiX) [inline graviti/operation/frame.py:142]: Will `dataframe.schema` always be record?

File: graviti/operation/frame.py [hunk 3]
```diff
@@ -134,3 +163,29 @@ def do(  # pylint: disable=invalid-name
...
+    arrays = []
+    for key in _get_file_column_keys(schema):
+        array: "Container" = dataframe
+        for subkey in key:
+            array = array[subkey]  # type: ignore[index]
+        arrays.append(array)
+    return arrays
+
+
+def _get_file_column_keys(schema: record) -> List[Tuple[str, ...]]:
```
Comment (linjiX) [inline graviti/operation/frame.py:178]: Shall we make this function more common, make it to search any type not only File from schema?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/operation/frame.py (hunk 1 above)
Comment (Lee-000) [reply graviti/operation/frame.py:145]: Do you mean uploading sheet data after done uploading all the binary files? 
I think that would make the upload operation seem slow.

Comment (linjiX) [reply graviti/operation/frame.py:145]: OK we can refine it afterwards

File: graviti/operation/frame.py [hunk 4]
```diff
@@ -134,3 +163,31 @@ def do(  # pylint: disable=invalid-name
...
+    arrays = []
+    for key in _get_target_type_keys(dataframe.schema, target_type_name):
+        array: "Container" = dataframe
+        for subkey in key:
+            array = array[subkey]  # type: ignore[index]
+        arrays.append(array)
+    return arrays
+
+
+def _get_target_type_keys(schema: PortexType, target_type_name: str) -> List[Tuple[str, ...]]:
```
Comment (linjiX) [inline graviti/operation/frame.py:178]: should this method in `PortexType`？

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/portex/builtin.py [hunk 5]
```diff
@@ -337,28 +337,34 @@ def to_pyarrow(self) -> pa.DataType:
         """
         return pa.struct(self.fields.to_pyarrow())  # pylint: disable=no-member
 
-    def _get_keys(self) -> List[Tuple[str, ...]]:
-        keys = []
+    def get_keys(self, target_type_name: Optional[str] = None) -> List[Tuple[str, ...]]:
```
Comment (linjiX) [inline graviti/portex/builtin.py:340]: `target_type_name` -> `type_name`?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by graczhual — APPROVED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**